pkg/operator/csidriveroperator/hypershift_deployment_controller.go

```diff
@@ -0,0 +1,183 @@
+package csidriveroperator
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	configv1listers "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/cluster-storage-operator/pkg/csoclients"
+	"github.com/openshift/cluster-storage-operator/pkg/operator/configobservation/util"
+	"github.com/openshift/cluster-storage-operator/pkg/operator/csidriveroperator/csioperatorclient"
+	csoutils "github.com/openshift/cluster-storage-operator/pkg/utils"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"github.com/openshift/library-go/pkg/operator/status"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"k8s.io/klog/v2"
+)
+
+// This HyperShiftDeploymentController installs and syncs CSI driver operator Deployment.
+// It replace ${LOG_LEVEL} in the Deployment with current log level.
+// It replaces images in the Deployment using  CSIOperatorConfig.ImageReplacer.
+// It produces following Conditions:
+// <CSI driver name>CSIDriverOperatorDeploymentProgressing
+// <CSI driver name>CSIDriverOperatorDeploymentDegraded
+// This controller doesn't set the Available condition to avoid prematurely cascading
+// up to the clusteroperator CR a potential Available=false. On the other hand it
+// does a better in making sure the Degraded condition is properly set if the
+// Deployment isn't healthy.
+type HyperShiftDeploymentController struct {
+	name              string
+	guestClient       *csoclients.Clients
+	mgmtClient        *csoclients.Clients
+	controlNamespace  string
+	operatorClient    v1helpers.OperatorClient
+	csiOperatorConfig csioperatorclient.CSIOperatorConfig
+	versionGetter     status.VersionGetter
+	targetVersion     string
+	eventRecorder     events.Recorder
+	infraLister       configv1listers.InfrastructureLister
+	factory           *factory.Factory
+}
+
+var _ factory.Controller = &HyperShiftDeploymentController{}
+
+var (
+	envHyperShiftImage = os.Getenv("HYPERSHIFT_IMAGE")
+)
+
+func NewHyperShiftControllerDeployment(
```

Comment:
If we remove `if !olmRemovalComplete {...}` from CSIDriverOperatorDeploymentController, then both HyperShiftControllerDeployment + CSIDriverOperatorDeploymentController are just library-go DeploymentControllers with extra  hooks.

IMO, we can safely remove OLMOperatorRemovalController now. and use DeploymentController with hooks. The code will get much simpler - *DriverStarter will just start DeploymentControllers with the right hooks.